# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../clanlib06/gcc41.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -Nur ClanLib-0.6.5-orig/Sources/Core/IOData/Unix/directory_scanner_unix.cpp ClanLib-0.6.5/Sources/Core/IOData/Unix/directory_scanner_unix.cpp
--- ClanLib-0.6.5-orig/Sources/Core/IOData/Unix/directory_scanner_unix.cpp	2002-01-10 10:23:30.000000000 +0000
+++ ClanLib-0.6.5/Sources/Core/IOData/Unix/directory_scanner_unix.cpp	2006-04-22 17:27:33.377138250 +0000
@@ -18,6 +18,7 @@
 #include <libgen.h>
 #include <fnmatch.h>
 #include "directory_scanner_unix.h"
+#include <assert.h>
 
 CL_DirectoryScanner_Unix::CL_DirectoryScanner_Unix ()
 	: dir_temp (NULL), entry (NULL)
diff -Nur ClanLib-0.6.5-orig/Sources/GUI/messagebox_generic.h ClanLib-0.6.5/Sources/GUI/messagebox_generic.h
--- ClanLib-0.6.5-orig/Sources/GUI/messagebox_generic.h	2002-01-16 19:16:51.000000000 +0000
+++ ClanLib-0.6.5/Sources/GUI/messagebox_generic.h	2006-04-22 17:33:27.303257250 +0000
@@ -26,7 +26,7 @@
 		const std::string &button2,
 		const std::string &button3);
 
-	CL_MessageBox_Generic::~CL_MessageBox_Generic();
+	~CL_MessageBox_Generic();
 
 	void set_text(const std::string &text);
 
diff -Nur ClanLib-0.6.5-orig/Sources/SmallJPEG/jpgd/jpegdecoder.h ClanLib-0.6.5/Sources/SmallJPEG/jpgd/jpegdecoder.h
--- ClanLib-0.6.5-orig/Sources/SmallJPEG/jpgd/jpegdecoder.h	2001-05-07 08:36:59.000000000 +0000
+++ ClanLib-0.6.5/Sources/SmallJPEG/jpgd/jpegdecoder.h	2006-04-22 17:41:24.493079750 +0000
@@ -493,26 +493,26 @@
 
   void find_eoi(void);
 //------------------
-  /*inline*/ uint jpeg_decoder::rol(uint i, uchar j);
-  /*inline*/ uint jpeg_decoder::get_char(void);
-  /*inline*/ uint jpeg_decoder::get_char(bool *Ppadding_flag);
-  /*inline*/ void jpeg_decoder::stuff_char(uchar q);
-  /*inline*/ uchar jpeg_decoder::get_octet(void);
-  /*inline*/ uint jpeg_decoder::get_bits_1(int num_bits);
-  /*inline*/ uint jpeg_decoder::get_bits_2(int numbits);
-  /*inline*/ int jpeg_decoder::huff_decode(Phuff_tables_t Ph);
+  /*inline*/ uint rol(uint i, uchar j);
+  /*inline*/ uint get_char(void);
+  /*inline*/ uint get_char(bool *Ppadding_flag);
+  /*inline*/ void stuff_char(uchar q);
+  /*inline*/ uchar get_octet(void);
+  /*inline*/ uint get_bits_1(int num_bits);
+  /*inline*/ uint get_bits_2(int numbits);
+  /*inline*/ int huff_decode(Phuff_tables_t Ph);
 #ifdef SUPPORT_X86ASM
-  /*inline*/ uint jpeg_decoder::huff_extend(uint i, int c);
+  /*inline*/ uint huff_extend(uint i, int c);
 #endif
-  /*inline*/ uchar jpeg_decoder::clamp(int i);
+  /*inline*/ uchar clamp(int i);
 
 #ifdef SUPPORT_MMX
-  /*inline*/ uint jpeg_decoder::get_high_byte_mmx(void);
-  /*inline*/ uint jpeg_decoder::get_high_word_mmx(void);
-  /*inline*/ void jpeg_decoder::get_bits_2_mmx_init(void);
-  /*inline*/ void jpeg_decoder::get_bits_2_mmx_deinit(void);
-  /*inline*/ uint jpeg_decoder::get_bits_2_mmx(int numbits);
-  /*inline*/ int jpeg_decoder::huff_decode_mmx(Phuff_tables_t Ph);
+  /*inline*/ uint get_high_byte_mmx(void);
+  /*inline*/ uint get_high_word_mmx(void);
+  /*inline*/ void get_bits_2_mmx_init(void);
+  /*inline*/ void get_bits_2_mmx_deinit(void);
+  /*inline*/ uint get_bits_2_mmx(int numbits);
+  /*inline*/ int huff_decode_mmx(Phuff_tables_t Ph);
 #endif
 //------------------
   int   image_x_size;
